alkira/resource_alkira_connector_gcp_interconnect.go

```diff
@@ -0,0 +1,341 @@
+				Description:  "The CXP where the connector should be provisioned.",
+				Type:         schema.TypeString,
+				Required:     true,
+				ValidateFunc: validation.StringInSlice([]string{"US-EAST1", "US-WEST1", "US-CENTRAL1", "AP-NORTHEAST", "EU-WEST"}, false),
+			},
+			"enabled": {
+				Description: "Is the connector enabled. Default is `true`.",
+				Type:        schema.TypeBool,
+				Optional:    true,
+				Default:     true,
+			},
+			"group": {
+				Description: "The group of the connector.",
+				Type:        schema.TypeString,
+				Optional:    true,
+			},
+			"tunnel_protocol": {
+				Description: "The tunnel protocol used by the connector." +
+					"Can be one of `GRE`, `IPSEC`, `VXLAN`, `VXLAN_GPE`.",
+				Type:         schema.TypeString,
+				Required:     true,
+				ValidateFunc: validation.StringInSlice([]string{"GRE", "IPSEC", "VXLAN", "VXLAN_GPE"}, false),
+			},
+			"billing_tag_ids": {
+				Description: "Billing tags to be associated with " +
+					"the resource. (see resource `alkira_billing_tag`).",
+				Type:     schema.TypeSet,
+				Optional: true,
+				Elem:     &schema.Schema{Type: schema.TypeInt},
+			},
+			"loopback_prefixes": {
+				Description: "A list of prefixes that should be " +
+					"associated with the connector.",
+				Type:     schema.TypeSet,
+				Required: true,
+				Elem:     &schema.Schema{Type: schema.TypeString},
+			},
+			"instance": {
+				Description: "A list of instances of the Interconnect",
+				Type:        schema.TypeSet,
+				Required:    true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"id": {
+							Description: "The ID of the instance.",
+							Type:        schema.TypeInt,
+							Optional:    true,
+							Computed:    true,
+						},
+						"name": {
+							Description: "The name of the instance.",
+							Type:        schema.TypeString,
+							Required:    true,
+						},
+						"edge_availibility_domain": {
+							Description:  "The Availibility Domain of the instance.",
+							Type:         schema.TypeString,
+							Required:     true,
+							ValidateFunc: validation.StringInSlice([]string{"AVAILABILITY_DOMAIN_ANY", "AVAILIBILITY_DOMAIN_1", "AVAILBILITY_DOMAIN_2"}, false),
+						},
+						"candidate_subnets": {
+							Description: "A list of candidate subnets to be used by the instance.",
+							Type:        schema.TypeSet,
+							Elem:        &schema.Schema{Type: schema.TypeString},
+						},
+						"customer_asn": {
+							Description: "The customer ASN.",
+							Type:        schema.TypeInt,
+							Required:    true,
+						},
+						"bgp_auth_key": {
+							Description: "The BGP MD5 authentication key to authenticate Alkira CXP.",
+							Type:        schema.TypeString,
+							Optional:    true,
+						},
+						"gateway_mac_address": {
+							Description: "The MAC address of the gateway." +
+								"Required if tunnel protocol is VXLAN.",
+							Type:     schema.TypeString,
+							Optional: true,
+						},
+						"vni": {
```

Comment:
double-check on this one... I don't think I saw GCP use this term before... so not entirely sure what is this ID. If it's truly an ID, probably vni_id makes more sense?